torchgen/autoheuristic/gen_data_pad_mm.py

```diff
@@ -0,0 +1,208 @@
+            return torch.mm(a, b)
+
+        def mm_mat1_prepadded(a, b):
+            return torch.mm(a + 1, b)
+
+        def mm_mat2_prepadded(a, b):
+            return torch.mm(a, b + 1)
+
+        def mm_mat1_mat2_prepadded(a, b):
+            return torch.mm(a + 1, b + 1)
+
+        if prepadded_left and prepadded_right:
+            cf = torch.compile(mm_mat1_mat2_prepadded)
+        elif prepadded_left:
+            cf = torch.compile(mm_mat1_prepadded)
+        elif prepadded_right:
+            cf = torch.compile(mm_mat2_prepadded)
+        else:
+            cf = torch.compile(mm)
+        print(f"{do_bench(lambda: cf(a, b))} ms")
+        torch.compiler.reset()
+
+
+def get_alignment_size(dtype: Any) -> int:
+    if dtype == torch.float16 or dtype == torch.half or dtype == torch.bfloat16:
+        return 8
+    elif dtype == torch.float32 or dtype == torch.float:
+        return 4
+    else:
+        return 0
+
+
+def fits_in_memory(dtype: Any, m: int, k: int, n: int):
+    if dtype == torch.float16 or dtype == torch.bfloat16:
+        return 2 * (m * k + k * n + m * n) < threshold_memory
+    elif dtype == torch.float32:
+        return 4 * (m * k + k * n + m * n) < threshold_memory
+
+
+def get_random_dim(min_power2: int = 1, max_power2: int = 16) -> int:
+    aligned = random.choices([True, False], [p_unaligned, 1 - p_unaligned])[0]
+    if aligned:
+        return 2 ** random.randint(min_power2, max_power2)
+    else:
+        # choose a random number between 2^i and 2^(i+1)
+        i = random.randint(min_power2, max_power2 - 1)
+        lower = 2**i + 1
+        upper = 2 ** (i + 1) - 1
+        return random.randint(lower, upper)
+
+
+def get_m_k_n(dtype: Any) -> Tuple[int, int, int]:
+    uniform = random.choices([True, False], [0.5, 0.5])[0]
+
+    # repeat until tensors fit in memory
+    while True:
+        if uniform:
+            m = random.randint(1, 65536)
+            k = random.randint(1, 65536)
+            n = random.randint(1, 65536)
+        else:
+            m = get_random_dim()
+            k = get_random_dim()
+            n = get_random_dim()
+
+        if fits_in_memory(dtype, m, k, n):
+            return (m, k, n)
+
+
+def transposed() -> bool:
+    return random.choices([True, False], [p_transposed, 1 - p_transposed])[0]
+
+
+def prepadded() -> bool:
+    return random.choices([True, False], [p_prepadded, 1 - p_prepadded])[0]
+
+
+def get_dtype() -> Any:
+    dtype_choices = [torch.float16, torch.bfloat16, torch.float32]
+    dtype_weights = [1, 1, 1]
+    return random.choices(dtype_choices, dtype_weights)[0]
+
+
+def set_precision(dtype: Any) -> None:
+    if dtype == torch.float32:
+        precisions = ["high", "highest"]
+        weights = [1 - p_float32_prec_highest, p_float32_prec_highest]
+        precision = random.choices(precisions, weights)[0]
+    else:
+        precision = "high"
+    torch.set_float32_matmul_precision(precision)
+
+
+def main() -> None:
+    while True:
```

Comment:
Instead of an infinite loop, maybe have an argument specifying how many samples we want to collect and show a progress bar with the tqdm package.